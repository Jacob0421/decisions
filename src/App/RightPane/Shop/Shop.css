.shop-title{
    font-size: 40px;
    font-weight: bold;
}

.item-section{
    background: white;
    display: flex;
    flex-direction: column;
}
.card{
    width: 100%;
    outline: 1px solid black;
    align-items: center;
}

.shop-divider {
    width: 100%;
    border-top: 2px solid grey;
}

.content{
    width:100%;
    max-height: 100%;
    background: brown;
}
.worker-details {
    display: flex;
    flex-direction: column;
    gap: 2vh
}