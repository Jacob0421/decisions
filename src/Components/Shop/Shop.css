.item-section{
    width: 80%;
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
    padding: 1vh 1vw;
    gap: 1vw;
    margin: 0 auto
}
.card{
    width: 20%;
    outline: 1px solid black;
    align-items: center;
}

.shop-divider {
    width: 100%;
    border-top: 2px solid grey;
}