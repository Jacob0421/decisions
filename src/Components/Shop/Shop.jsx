import React from "react";

export default function Shop() {
	return (
		<>
			<h1>Shop</h1>
			{/* <button onClick={handleNewWorker("Reaper")}></button> */}
		</>
	);
}
