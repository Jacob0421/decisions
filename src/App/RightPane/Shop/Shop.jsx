import React from "react";
import { Upgrades, Consumables } from "../../../Data/ShopItems";
import ShopItems from "./ShopItems/ShopItems";
import "./Shop.css";

export default function Shop(params) {
	const { handleBuy } = params;

	return (
		<div className="content center-text">
			<span className="title">Shop</span>

			<div className="shop-divider"></div>

			<h2 className="section-header">Upgrades</h2>
			<div className="item-section">
				{Upgrades.map((item, index) => (
					<ShopItems
						Key={index}
						id={index}
						itemName={item.name}
						itemCost={item.cost}
						itemEffect={item.effect}
						levelAffected={item.levelAffected}
						buildingAffected={item.buildingAffected}
						upgradeType={item.upgradeType}
						upgradeModifier={item.upgradeModifier}
						handleBuy={handleBuy}
					/>
				))}
			</div>
			<div className="shop-divider"></div>
			<h2 className="section-header">Consumables</h2>
			<div className="item-section">
				{Consumables.map((item, index) => (
					<button className="card flex col">
						<p>{item.name}</p>
						<label>
							<strong>Effect:</strong>
							<p>{item.effect}</p>
						</label>
						<p>Cost: {item.cost}</p>
					</button>
				))}
			</div>
			<div className="shop-divider"></div>
		</div>
	);
}
