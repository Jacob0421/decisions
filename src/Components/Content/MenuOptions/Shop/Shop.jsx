import React from "react";
import { Upgrades, Consumables } from "../../../../Data/ShopItems";
import "./Shop.css";

export default function Shop() {
	return (
		<div className="content center-text">
			<span className="title">Shop</span>

			<div className="shop-divider"></div>

			<h2 className="section-header">Upgrades</h2>
			<div className="item-section">
				{Upgrades.map((item, index) => (
					<button className="card flex col">
						<p>{item.name}</p>
						<label>
							<strong>Effect:</strong>
							<p>{item.effect}</p>
						</label>
						<p>Cost: {item.cost}</p>
					</button>
				))}
			</div>
			<div className="shop-divider"></div>
			<h2 className="section-header">Consumables</h2>
			<div className="item-section">
				{Consumables.map((item, index) => (
					<button className="card flex col">
						<p>{item.name}</p>
						<label>
							<strong>Effect:</strong>
							<p>{item.effect}</p>
						</label>
						<p>Cost: {item.cost}</p>
					</button>
				))}
			</div>
			<div className="shop-divider"></div>
		</div>
	);
}
