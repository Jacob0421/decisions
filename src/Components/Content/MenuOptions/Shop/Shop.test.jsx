import { render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Shop from './Shop';

afterEach(() => {
    cleanup();
});


test('shop', () => {
    render(<Shop />);
    const title = screen.getByText(/shop/i);
    expect(title).toBeInTheDocument();
    //expect(true).toBe(true);
});

test('shop2', () => {
    render(<Shop />);
    const shopElement = screen.getByTestId('shop-1');
   // const title = screen.getByText(/shop/);
    expect(shopElement).toBeInTheDocument();
    expect(shopElement).toHaveTextContent('Shop');
});

test('matches snapshot', () => {
    const tree = renderer.create(<Shop />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
});
