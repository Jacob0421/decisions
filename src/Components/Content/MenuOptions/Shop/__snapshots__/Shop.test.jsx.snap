// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="center-container center-text"
  data-testID="shop-1"
>
  <h1>
    Shop
  </h1>
  <div
    className="content"
  >
    <h2
      className="section-header"
    >
      Upgrades
    </h2>
    <div
      className="item-section"
    >
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
    </div>
    <div
      className="shop-divider"
    />
    <h2
      className="section-header"
    >
      Consumables
    </h2>
    <div
      className="item-section"
    >
      <button
        className="card flex col"
      >
        <p>
          Additional Reapers
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            +1 Reapers
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
      <button
        className="card flex col"
      >
        <p>
          Mo' Money, Mo' Problems
        </p>
        <label>
          <strong>
            Effect:
          </strong>
          <p>
            Reapers generate more money when harvesting souls
          </p>
        </label>
        <p>
          Cost: 
          100
        </p>
      </button>
    </div>
    <div
      className="shop-divider"
    />
  </div>
</section>
`;
